### PR TITLE
Support chains of piped commands in dockerCall

### PR DESCRIPTION
**Pull request by glennhickey**
Current interface not affected, but if parameters argument is a list of lists, instead of a list, then it will be treated as chain of piped commands.  

This is based on  work by @cmarkello here: https://github.com/cmarkello/toil-lib/commit/75c21593abb1c5856d5e4be01ef5c1018cc13104 and is required for toil-vg, where we often pipe a bunch of vg commands together.

### CODE REVIEW

**Comment by glennhickey** [comment]
Um, can any toilers help me with this error (from jenkins-it)?  Retribution for criticizing the multipart uploadinger in another thread?

  File "/usr/local/lib/python2.7/dist-packages/boto/s3/bucket.py", line 1765, in initiate_multipart_upload
    response.status, response.reason, body)
S3ResponseError: S3ResponseError: 403 Forbidden

File: src/toil/lib/docker.py [hunk 1]
```diff
@@ -148,7 +151,18 @@ def _docker(job,
...
     # Make subprocess call
-    call = baseDockerCall + [tool] + parameters
+
+    # If parameters is list of lists, treat each list as separate command and chain with pipes
+    if len(parameters) > 0 and type(parameters[0]) is list:
+        # When piping, all arguments get passed as a single string to bash -c.
+        # We try to support spaces in paths by wrapping them all in quotes first.
+        quoted_params = []
+        for params in parameters:
+            quoted_params.append(['\'{}\''.format(p) for p in params])
```
Comment (cket) [inline src/toil/lib/docker.py]: @glennhickey this might help you out https://docs.python.org/2/library/pipes.html#pipes.quote

**Comment by cket** [comment]
This should also have test(s) showing that the chaining works as expected

**Comment by glennhickey** [comment]
@cket test added.  pipes.quote() used.

**Comment by jvivian** [comment]
Elegant implementation — LG2M!

File: src/toil/test/lib/dockerTest.py [hunk 2]
```diff
@@ -94,6 +94,20 @@ def testDockerClean(self):
                         _dockerKill(container_name, RM)
                         os.remove(test_file)
 
+    def testDockerPipeChain(self):
+        """
+        Test for piping API for dockerCall().  Using this API (activated when list of
+        argument lists is given as paramters), commands a piped together into a chain
```
Comment (cket) [inline src/toil/test/lib/dockerTest.py]: *parameters

File: src/toil/lib/docker.py [hunk 3]
```diff
@@ -36,6 +36,7 @@ def dockerCall(job,
     :param toil.Job.job job: The Job instance for the calling function.
     :param str tool: Name of the Docker image to be used (e.g. quay.io/ucsc_cgl/samtools:latest).
     :param list[str] parameters: Command line arguments to be passed to the tool.
+           If list of lists: list[list[str]], then treat as succesive commands chained with pipe.
```
Comment (cket) [inline src/toil/lib/docker.py]: *successive. This occurs in the 2 other doc strings as well.

**Comment by benedictpaten** [comment]
Let's get this in. @fnothaft any comment?

**Comment by fnothaft** [comment]
@benedictpaten will look this PM

**Comment by cket** [comment]
Great, thanks @glennhickey and @fnothaft

**Pull request merged**